sklearn/tree/_tree.pyx

```diff
@@ -2707,12 +2707,11 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                                          split.threshold, impurity, n_node_samples,
                                          weighted_n_node_samples)
 
-                if is_leaf:
-                    # Don't store value for internal nodes
-                    splitter.node_value(tree.value +
-                                        node_id * tree.value_stride)
+                # Store value for all nodes, to facilitate tree/model inspection and interpretation
+                splitter.node_value(tree.value
+                                    + node_id * tree.value_stride)
```

Comment:
This should be done in a separate pr with appropriate benchmarks as discussed in #4644.